aws/resource_aws_serverlessrepository_stack.go

```diff
@@ -0,0 +1,321 @@
+	changeSetRequest := serverlessrepository.CreateCloudFormationChangeSetRequest{
+		StackName:     aws.String(stackName),
+		ApplicationId: aws.String(d.Get("application_id").(string)),
+		Capabilities:  expandStringSet(d.Get("capabilities").(*schema.Set)),
+		Tags:          keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreServerlessApplicationRepository().ServerlessapplicationrepositoryTags(),
+	}
+	if v, ok := d.GetOk("semantic_version"); ok {
+		version := v.(string)
+		changeSetRequest.SemanticVersion = aws.String(version)
+	}
+	if v, ok := d.GetOk("parameters"); ok {
+		changeSetRequest.ParameterOverrides = expandServerlessRepositoryChangeSetParameters(v.(map[string]interface{}))
+	}
+
+	log.Printf("[DEBUG] Creating Serverless Application Repository change set: %s", changeSetRequest)
+	changeSetResponse, err := serverlessConn.CreateCloudFormationChangeSet(&changeSetRequest)
+	if err != nil {
+		return fmt.Errorf("error creating Serverless Application Repository change set (%s): %w", stackName, err)
+	}
+
+	d.SetId(aws.StringValue(changeSetResponse.StackId))
+
+	_, err = cfwaiter.ChangeSetCreated(cfConn, d.Id(), aws.StringValue(changeSetResponse.ChangeSetId))
+	if err != nil {
+		return fmt.Errorf("error waiting for Serverless Application Repository change set (%s) creation: %w", stackName, err)
+	}
+
+	log.Printf("[INFO] Serverless Application Repository change set (%s) created", d.Id())
+
+	requestToken := resource.UniqueId()
+	executeRequest := cloudformation.ExecuteChangeSetInput{
+		ChangeSetName:      changeSetResponse.ChangeSetId,
+		ClientRequestToken: aws.String(requestToken),
+	}
+	log.Printf("[DEBUG] Executing Serverless Application Repository change set: %s", executeRequest)
+	_, err = cfConn.ExecuteChangeSet(&executeRequest)
+	if err != nil {
+		return fmt.Errorf("Executing Serverless Application Repository change set failed: %w", err)
+	}
+
+	_, err = cfwaiter.StackCreated(cfConn, d.Id(), requestToken, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return fmt.Errorf("error waiting for Serverless Application Repository CloudFormation Stack (%s) creation: %w", d.Id(), err)
+	}
+
+	log.Printf("[INFO] Serverless Application Repository CloudFormation Stack (%s) created", d.Id())
+
+	return resourceAwsServerlessRepositoryStackRead(d, meta)
+}
+
+func resourceAwsServerlessRepositoryStackRead(d *schema.ResourceData, meta interface{}) error {
+	serverlessConn := meta.(*AWSClient).serverlessapplicationrepositoryconn
+	cfConn := meta.(*AWSClient).cfconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	getApplicationInput := &serverlessrepository.GetApplicationInput{
+		ApplicationId: aws.String(d.Get("application_id").(string)),
+	}
+
+	_, ok := d.GetOk("semantic_version")
+	if !ok {
+		getApplicationOutput, err := serverlessConn.GetApplication(getApplicationInput)
+		if err != nil {
+			return err
+		}
+		d.Set("semantic_version", getApplicationOutput.Version.SemanticVersion)
+	}
```

Comment:
This will likely cause issues on import since it is currently set to `ImportStatePassthrough` -- will need to add these to custom import function where they are present in the ID then passed into the correct attributes for the `Read` function.